File: bar_backend/src/main/java/com/gbyzzz/bar_web_app/bar_backend/dto/CocktailRecipeDTO.java
package com.gbyzzz.bar_web_app.bar_backend.dto;

import com.fasterxml.jackson.annotation.JsonInclude;
import com.fasterxml.jackson.annotation.JsonProperty;
import com.fasterxml.jackson.annotation.JsonTypeInfo;
import jakarta.persistence.Id;

import java.util.List;

//@JsonTypeInfo(use = JsonTypeInfo.Id.CLASS, include = JsonTypeInfo.As.PROPERTY, property = "@class")
@JsonInclude(JsonInclude.Include.NON_NULL)
public record CocktailRecipeDTO(
        @JsonProperty("cocktailDTO")
        CocktailDTO cocktailDTO,
        @JsonProperty("recipesDTO")
        List<RecipeDTO> recipesDTO
) {

}

File: bar_frontend/src/app/model/dto/CocktailRecipeDTO.ts
import {Cocktail} from "../Cocktail";
import {Recipe} from "../Recipe";

export class CocktailRecipeDTO{
  cocktailDTO: Cocktail;
  recipesDTO: Recipe[];


  constructor(cocktailDTO?: Cocktail, recipesDTO?: Recipe[]) {
    this.cocktailDTO = cocktailDTO;
    this.recipesDTO = recipesDTO;
  }
  // toJSON(): any {
  //   const { cocktailDTO, ...rest } = this;
  //   return {
  //     '@class': 'com.gbyzzz.bar_web_app.bar_backend.dto.CocktailRecipeDTO',
  //     'cocktailDTO': this.cocktailDTO,
  //     'recipesDTO': this.recipesDTO
  //   };
  // }
}
